git/git.go

```diff
@@ -110,41 +113,40 @@ func Middleware(repoDir string, gh Hooks) wish.Middleware {
 	}
 }
 
-func gitReceivePack(s ssh.Session, gitCmd string, repoDir string, repo string) error {
-	err := ensureRepo(repoDir, repo)
-	if err != nil {
-		return err
-	}
-	rp := filepath.Join(repoDir, repo)
-	err = runCmd(s, "./", gitCmd, rp)
-	if err != nil {
-		return err
-	}
-	err = ensureDefaultBranch(s, rp)
-	if err != nil {
-		return err
-	}
-	err = runCmd(s, rp, "git", "update-server-info")
-	if err != nil {
-		return err
-	}
-	return nil
-}
-
-func gitUploadPack(s ssh.Session, gitCmd string, repoDir string, repo string) error {
+func gitPack(s ssh.Session, gitCmd string, repoDir string, repo string) error {
+	cmd := strings.TrimPrefix(gitCmd, "git-")
 	rp := filepath.Join(repoDir, repo)
-	exists, err := fileExists(rp)
-	if !exists {
-		return ErrInvalidRepo
-	}
-	if err != nil {
-		return err
-	}
-	err = runCmd(s, "./", gitCmd, rp)
-	if err != nil {
-		return err
+	switch gitCmd {
+	case "git-upload-archive", "git-upload-pack":
+		exists, err := fileExists(rp)
+		if !exists {
+			return ErrInvalidRepo
+		}
+		if err != nil {
+			return err
+		}
+		return runCmd(s, "./", "git", cmd, rp)
+	case "git-receive-pack":
+		err := ensureRepo(repoDir, repo)
+		if err != nil {
+			return err
+		}
+		err = runCmd(s, "./", "git", cmd, rp)
+		if err != nil {
+			return err
+		}
+		err = ensureDefaultBranch(s, rp)
+		if err != nil {
+			return err
+		}
+		err = runCmd(s, rp, "git", "update-server-info")
+		if err != nil {
+			return err
+		}
+		return nil
```

Comment:
could be simplified a bit:
```suggestion
		if err := ensureRepo(repoDir, repo); err != nil {
			return err
		}
		if err := runCmd(s, "./", "git", cmd, rp); err != nil {
			return err
		}
		if err := ensureDefaultBranch(s, rp); err != nil {
			return err
		}
		return runCmd(s, rp, "git", "update-server-info")
```